src/daemon/normalize.js

```diff
@@ -0,0 +1,12 @@
+const punycode = require('punycode') // eslint-disable-line node/no-deprecated-api
+
+module.exports = function normalize(id) {
+  return id
+    ? punycode
+        .encode(id)
```

Comment:
punycode won't mess with existing ASCII; it only ensures that non-ASCII characters get embedded into a hostname-valid space.